backend/typescript/services/implementations/campService.ts

```diff
@@ -1,17 +1,85 @@
-import { CreateCampDTO, CampDTO, CamperCSVInfoDTO } from "../../types";
+import {
+  CreateCampDTO,
+  CampDTO,
+  CamperCSVInfoDTO,
+  GetCampDTO,
+  FormQuestionDTO,
+} from "../../types";
 import ICampService from "../interfaces/campService";
 import MgCamp, { Camp } from "../../models/camp.model";
 import MgCamper, { Camper } from "../../models/camper.model";
 import { getErrorMessage } from "../../utilities/errorUtils";
 import { generateCSV } from "../../utilities/CSVUtils";
 import logger from "../../utilities/logger";
 import MgBaseCamp, { BaseCamp } from "../../models/baseCamp.model";
-import MgFormQuestion from "../../models/formQuestion.model";
+import MgFormQuestion, { FormQuestion } from "../../models/formQuestion.model";
 
 const Logger = logger(__filename);
 
 class CampService implements ICampService {
   /* eslint-disable class-methods-use-this */
+  async getCamps(): Promise<GetCampDTO[]> {
+    try {
+      const camps: Camp[] | null = await MgCamp.find({}).populate({
+        path: "baseCamp",
+        model: MgBaseCamp,
+        populate: {
+          path: "formQuestions",
+          model: MgFormQuestion,
+        },
+      });
+
+      if (!camps) {
+        return [];
+      }
+
+      return camps.map((camp) => {
+        const baseCamp = camp.baseCamp as BaseCamp;
```

Comment:
I think there might be a compiler error here?

<img width="1279" alt="image" src="https://user-images.githubusercontent.com/31112327/159110704-b9e9ce18-42de-49a8-9beb-35a022e75dc9.png">

Reply:
hmm, I don't see it from my end. Are you sure you have `baseCamp: BaseCamp | Schema.Types.ObjectId;` and not `baseCamp: Schema.Types.ObjectId;` in your camp.model.ts. That change is included in my commit https://github.com/uwblueprint/focus-on-nature/blob/feat/create-get-camp-endpoint/backend/typescript/models/camp.model.ts#L6

Reply:
ah you're right sorry – the merge conflict didn't show up in gh desktop..

I was looking through your PR to see what needs to be updated since we're refactoring the baseCamp/Camp model, hope I didn't wake you up

Reply:
okk, no worries. I usually end up sleeping pretty late Friday nights so all good :)